momento-sdk/src/main/java/momento/sdk/IAuthClient.java

```diff
@@ -0,0 +1,15 @@
+package momento.sdk;
+
+import java.util.concurrent.CompletableFuture;
+import momento.sdk.auth.accessControl.DisposableTokenScope;
+import momento.sdk.auth.accessControl.ExpiresIn;
+import momento.sdk.responses.GenerateDisposableTokenResponse;
+
+public interface IAuthClient extends AutoCloseable {
```

Comment:
we are using AutoCloseable here, but just `Closeable` on the `CacheClient`. This seems to cause a difference in how error handling works. Not quite sure if one is better than the other, maybe @nand4011 has input.

Reply:
Seems like `AutoCloseable` is more modern/preferred; perhaps we can see if anything breaks if we switch the main `CacheClient` over to that as well.